### PR TITLE
Fix: Ensure escope honors rest args (fixes #47)

### LINKED ISSUE #47: Rest arguments are ignored
In the latest escope with `ecmaVersion: 6`, rest arguments are ignored. For example:

```
function f(...b) {
}
```

In this code, the `b` argument isn't found by escope.

I believe I have a fix for this. 

Related: https://github.com/eslint/eslint/issues/1543

### PR DESCRIPTION
**Pull request by nzakas**

### CODE REVIEW

File: test/es6-destructuring-assignments.coffee [hunk 1]
```diff
@@ -576,11 +576,12 @@ describe 'ES6 destructuring assignments', ->
 
         scope = scopeManager.scopes[1]
         expect(scope.type).to.be.equal 'function'
-        expect(scope.variables).to.have.length 4
+        expect(scope.variables).to.have.length 5
```
Comment (nzakas) [inline test/es6-destructuring-assignments.coffee]: This test was wrong previously, it missed the final argument.

**Comment by Constellation** [comment]
Looks very nice! I'll merge it.

**Comment by Constellation** [comment]
Is it preferable to ship it to 2.0.x version?

**Comment by Constellation** [comment]
I'll backport it.

**Comment by nzakas** [comment]
Yeah, in 2.0.x would be great.

**Comment by Constellation** [comment]
Now backported as `2.0.5` and released :)

**Comment by nzakas** [comment]
Thanks!

**Pull request merged**